cw-orch/src/mock/mock_querier.rs

```diff
@@ -0,0 +1,261 @@
+use crate::daemon::queriers::CosmWasm;
+use crate::daemon::queriers::DaemonQuerier;
+use crate::prelude::queriers::Bank;
+use crate::prelude::queriers::Staking;
+use cosmwasm_std::Addr;
+use cosmwasm_std::AllBalanceResponse;
+use cosmwasm_std::BalanceResponse;
+use cosmwasm_std::Delegation;
+use cosmwasm_std::{AllDelegationsResponse, BondedDenomResponse};
+
+use cosmwasm_std::BankQuery;
+use cosmwasm_std::Binary;
+use cosmwasm_std::Empty;
+use cosmwasm_std::StakingQuery;
+use ibc_chain_registry::chain::ChainData;
+use tokio::runtime::Runtime;
+use tonic::transport::Channel;
+
+use std::marker::PhantomData;
+use std::str::FromStr;
+
+use cosmwasm_std::testing::{MockApi, MockQuerier, MockStorage};
+use cosmwasm_std::{
+    from_slice, to_binary, Coin, ContractResult, OwnedDeps, Querier, QuerierResult, QueryRequest,
+    SystemError, SystemResult, Uint128, WasmQuery,
+};
+
+use crate::daemon::GrpcChannel;
+
+fn to_cosmwasm_coin(c: cosmrs::proto::cosmos::base::v1beta1::Coin) -> Coin {
+    Coin {
+        amount: Uint128::from_str(&c.amount).unwrap(),
+        denom: c.denom,
+    }
+}
+
+const QUERIER_ERROR: &str =
+    "Only Bank balances and Wasm (raw + smart) and Some staking queries are covered for now";
+
+/// mock_dependencies is a drop-in replacement for cosmwasm_std::testing::mock_dependencies
+/// this uses our CustomQuerier.
+pub fn mock_dependencies(
+    chain_info: ChainData,
+) -> OwnedDeps<MockStorage, MockApi, WasmMockQuerier> {
+    let custom_querier: WasmMockQuerier = WasmMockQuerier::new(MockQuerier::new(&[]), chain_info);
+
+    OwnedDeps {
+        storage: MockStorage::default(),
+        api: MockApi::default(),
+        querier: custom_querier,
+        custom_query_type: PhantomData,
+    }
+}
+
+/// Querier struct that fetches queries on-chain directly
+pub struct WasmMockQuerier {
+    base: MockQuerier<Empty>,
```

Comment:
is this base required anywhere? If not, is there a scenario where it might be? If yes, expose it with `pub` or a getter. If not you can remove it.

Reply:
https://github.com/AbstractSDK/cw-orchestrator/pull/147/commits/1398297fe7ba043e11ec874b01616c53f9536c38